File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int letters(string text);
int words(string text);
int sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    // counting letters, words and sentences using functions from the bottom of document and assigning them to variables
    int letters_count = letters(text);
    int words_count = words(text);
    int sentence_count = sentences(text);

    // finging L and S using variable above
    float L = (float) letters_count / (float) words_count * 100;
    float S = (float) sentence_count / (float) words_count * 100;

    // finding index by applying Coleman-Liau index formula
    int index = round((0.0588 * L) - (0.296 * S) - 15.8);

    // outputing the result
    if (index >= 16)
    {
        printf("Grade 16+\n");
    } 
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int letters(string text)
{
    int letter = 0;
    int text_len = strlen(text);

    for (int i = 0; i < text_len; i++)
    {
        int x = toupper(text[i]);
        if (x >= 65 && x <= 90)
        {
            letter++;
        }
    }

    return letter;
}

int words(string text)
{
    int words = 0;
    int text_len = strlen(text);

    for (int i = 0; i < text_len; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }

    words++;
    return words;
}

int sentences(string text)
{
    int sentence = 0;
    int text_len = strlen(text);

    for (int i = 0; i < text_len; i++)
    {
        char x = text[i];
        if (x == '.' || x == '!' || x == '?')
        {
            sentence++;
        }
    }

    return sentence;
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

# print index rounded to nearest int
# if index > 16 print grade 16+
# if index < 1 print before grade 1


def main():
    letter = letters(text)
    word = words(text)
    sentence = sentences(text)

    L = letter / word * 100
    S = sentence / word * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)

    if (index >= 16):
        print("Grade 16+")
    elif (index < 1):
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def letters(text):
    letter = 0
    for x in text:
        x = x.upper()
        if (ord(x) >= 65 and ord(x) <= 90):
            letter += 1
    return letter


def words(text):
    wordCount = 0

    for x in text:
        if (x == " "):
            wordCount += 1

    wordCount += 1
    return wordCount


def sentences(text):
    sentCount = 0
    for x in text:
        if (x == "." or x == "!" or x == "?"):
            sentCount += 1

    return sentCount


main()
